source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1342,10 +1457,21 @@ void RtFileIo::readLibraries(const FilePathVec& libraryPaths, const FileSearchPa
         stage->addSourceUri(RtToken(uri));
     }
 
+    // Load and register all targetdefs. Having these available is needed
+    // when implementations are loaded later.
+    for (const TargetDefPtr& targetdef : doc->getTargetDefs())
```

Comment:
This code looks similar to code above without the filter, is there some way to avoid the code duplication? Just a thought.

Reply:
Not sure if you saw this comment, just wanted to check.